### PR TITLE
update Rectangle to use Check

### PR DESCRIPTION
**Pull request by austinEng**
part of #4794

### CODE REVIEW

File: Source/Core/Rectangle.js [hunk 1]
```diff
@@ -227,9 +218,7 @@ define([
      */
     Rectangle.fromCartographicArray = function(cartographics, result) {
         //>>includeStart('debug', pragmas.debug);
-        if (!defined(cartographics)) {
-            throw new DeveloperError('cartographics is required.');
-        }
+        Check.typeOf.object(cartographics, 'cartographics');
```
Comment (lilleyse) [inline Source/Core/Rectangle.js]: `cartographics` is an array

File: Source/Core/Rectangle.js [hunk 2]
```diff
@@ -284,9 +273,7 @@ define([
      */
     Rectangle.fromCartesianArray = function(cartesians, ellipsoid, result) {
         //>>includeStart('debug', pragmas.debug);
-        if (!defined(cartesians)) {
-            throw new DeveloperError('cartesians is required.');
-        }
+        Check.typeOf.object(cartesians, 'cartesians');
```
Comment (lilleyse) [inline Source/Core/Rectangle.js]: Same for `cartesians`.

File: Source/Core/Rectangle.js [hunk 3]
```diff
@@ -428,45 +413,28 @@ define([
...
-        if (typeof north !== 'number') {
-            throw new DeveloperError('north is required to be a number.');
-        }
+        Check.typeOf.number(north, 'north');
+        Check.numeric.minimum(north, -CesiumMath.PI_OVER_TWO);
+        Check.numeric.maximum(north, CesiumMath.PI_OVER_TWO);
 
-        if (north < -CesiumMath.PI_OVER_TWO || north > CesiumMath.PI_OVER_TWO) {
-            throw new DeveloperError('north must be in the interval [-Pi/2, Pi/2].');
-        }
```
Comment (lilleyse) [inline Source/Core/Rectangle.js]: Remove this line of whitespace.

**Comment by austinEng** [comment]
Updated

**Comment by lilleyse** [comment]
Thanks @austinEng, looks good.

**Pull request merged**